### PR TITLE
Migration Guide

### LINKED ISSUE #11: Migration guide
Document:

- [x] breaking changes (dropped features and renaming)
- [x] new features
- [x] how to migrate code
- [x]  action proba removed for now

### PR DESCRIPTION
**Pull request by araffin**
## Description

This is a first draft of the migration guide. Most of the content is there but it should be probably refined in a follow-up PR before v1.0 release.

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes)

closes #11 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (update in the documentation)

## Checklist:

- [x] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [ ] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.
- [x] I have reformatted the code using `make format` (**required**)
- [x] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [x] I have ensured `make pytest` and `make type` both pass. (**required**)

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

File: docs/guide/migration.rst [hunk 1]
```diff
@@ -9,4 +9,184 @@ This is a guide to migrate from Stable-Baselines to Stable-Baselines3.
...
+- Dropped parameter noise for DDPG and DQN
+- PPO is now closer to the original implementation (no clipping of the value function by default), cf PPO section below
+- orthogonal initialization is only used by A2C/PPO
+- the features extractor (CNN extractor) is shared between policy and q-networks for DDPG/SAC/TD3 and only the policy loss used to update it (much faster)
+- Tensorboard legacy logging was dropped in favor of having one logger for the terminal and Tensorboard (cf :ref:`Tensorboard integration <tensorboard>`)
+- we dropped ACKTR/ACER support because of their complexity compared to simpler alternatives (PPO, SAC, TD3) performing as good.
+- we dropped GAIL support as we are focusing on model-free RL only, you can however take a look at the `Imitation Learning Baseline Implementations <https://github.com/HumanCompatibleAI/imitation>`_
+  which are based on SB3.
+
+TODO: change to deterministic predict for SAC/TD3
```
Comment (Miffyli) [inline docs/guide/migration.rst:43]: Not needed TODO?

File: docs/guide/migration.rst [hunk 2]
```diff
@@ -9,4 +9,184 @@ This is a guide to migrate from Stable-Baselines to Stable-Baselines3.
...
+
+state api breaking changes and implementation differences (e.g. clip range ppo and renaming of parameters)
```
Comment (Miffyli) [inline docs/guide/migration.rst:45]: Sounds like some TODO notes to me?

File: docs/guide/migration.rst [hunk 3]
```diff
@@ -9,4 +9,184 @@ This is a guide to migrate from Stable-Baselines to Stable-Baselines3.
...
+.. code-block:: python
+
+  from stable_baselines3.common.cmd_util import make_atari_env, make_vec_env
+  from stable_baselines3.common.utils import set_random_seed
+
+instead of ``from stable_baselines3.common import make_atari_env``
+
+
+
+Parameters Change and Renaming
```
Comment (Miffyli) [inline docs/guide/migration.rst:65]: Suggestion: "Changes and renaming in parameters"

File: docs/guide/migration.rst [hunk 4]
```diff
@@ -9,4 +9,184 @@ This is a guide to migrate from Stable-Baselines to Stable-Baselines3.
...
+Internal Changes
+----------------
+
+Please read the :ref:`Developper Guide <developer>` section.
+
+
+New Features
+============
+
+- much cleaner and consistent base code (and no more warnings =D!) and static type checks
```
Comment (Miffyli) [inline docs/guide/migration.rst:163]: Capital letters for consistency

File: docs/guide/migration.rst [hunk 5]
```diff
@@ -9,4 +9,184 @@ This is a guide to migrate from Stable-Baselines to Stable-Baselines3.
...
+
+How to migrate?
+===============
+
+In most cases, replacing ``from stable_baselines`` by ``from stable_baselines3`` will be sufficient.
+Some files were moved to the common folder (cf above) and could result to import errors.
+We recommend looking at the `rl-zoo3 <https://github.com/DLR-RM/rl-baselines3-zoo>`_ and compare the imports
+to the `rl-zoo <https://github.com/araffin/rl-baselines-zoo>`_ of SB2 to have a concrete example of successful migration.
+
+Planned Features
```
Comment (Miffyli) [inline docs/guide/migration.rst:186]: Not really needed here. We will forget to update this in any case. Better to only have it in one place (e.g. readme or the issues)

File: docs/guide/migration.rst [hunk 6]
```diff
@@ -9,4 +9,184 @@ This is a guide to migrate from Stable-Baselines to Stable-Baselines3.
...
+
+  - ``logkv`` -> ``record``, ``writekvs`` -> ``write``, ``writeseq`` ->  ``write_sequence``,
+  - ``logkvs`` -> ``record_dict``, ``dumpkvs`` -> ``dump``,
+  - ``getkvs`` -> ``get_log_dict``, ``logkv_mean`` -> ``record_mean``,
+
+
+Internal Changes
+----------------
+
+Please read the :ref:`Developper Guide <developer>` section.
```
Comment (Miffyli) [inline docs/guide/migration.rst:157]: Typo "developper"

File: docs/guide/migration.rst (hunk 2 above)
Comment (araffin) [reply docs/guide/migration.rst:45]: this is done in fact... will remove

File: docs/guide/migration.rst (hunk 1 above)
Comment (araffin) [reply docs/guide/migration.rst:43]: Good point, will state the change

File: docs/guide/migration.rst (hunk 6 above)
Comment (araffin) [reply docs/guide/migration.rst:157]: french spelling :see_no_evil:

File: docs/guide/migration.rst [hunk 7]
```diff
@@ -9,4 +9,193 @@ This is a guide to migrate from Stable-Baselines to Stable-Baselines3.
 
 It also references the main changes.
 
-**TODO**
+.. warning::
+	This section is still a Work In Progress (WIP) Things might be added in the future before 1.0 release.
```
Comment (Miffyli) [inline docs/guide/migration.rst:13]: I think this part could be removed as we are near the release and probably won't update this doc much after release?

File: docs/guide/migration.rst [hunk 8]
```diff
@@ -9,4 +9,193 @@ This is a guide to migrate from Stable-Baselines to Stable-Baselines3.
...
+  - ``getkvs`` -> ``get_log_dict``, ``logkv_mean`` -> ``record_mean``,
+
+
+Internal Changes
+----------------
+
+Please read the :ref:`Developer Guide <developer>` section.
+
+
+New Features
```
Comment (Miffyli) [inline docs/guide/migration.rst:186]: Could say "New Features (v1.0 versus stable-baselines)" to highlight that these are major changes compared to stable-baselines2. I do not think we will be updating this list later on.

**Comment by Miffyli** [comment]
Just two more comments I could see. After addressing those I think this is good to go finally!

**Review by Miffyli — APPROVED** [review]
If no more changes coming, LGTM!

**Pull request merged**